compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -8640,7 +8660,8 @@ public void visit(BLangReCharacterClass reCharacterClass) {
         reCharacterClass.negation = rewriteExpr(reCharacterClass.negation);
         // Create empty charSet.
         if (reCharacterClass.charSet == null) {
-            reCharacterClass.charSet = ASTBuilderUtil.createEmptyCharSet(symTable.anydataType);
+            result = ASTBuilderUtil.createLiteral(reCharacterClass.pos, symTable.stringType, "");
```

Comment:
I think this change is not valid now.
Team decided to not to go with this fix,
I think you can close this PR